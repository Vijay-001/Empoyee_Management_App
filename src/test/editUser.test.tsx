import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { mount, shallow } from 'enzyme';
import { TextField } from '@material-ui/core';
import { fireEvent, render } from '@testing-library/react';
import { shallowToJson } from 'enzyme-to-json';
import { Button, Modal } from 'react-bootstrap';
import UserEditModal from '../containers/editUser/edit';

const mockStore = configureMockStore();
const store = mockStore({});

describe('<AdminLogin/>', () => {
  const fieldprops = {
    first_name: '',
    id: '',
    last_name: '',
    email: '',
    avatar: '',
    password: '',
  };

  let wrapper = beforeEach(() => {
    wrapper = shallow(
      <Provider store={store}>
        <UserEditModal show onClose={() => false} mode="Edit" userInfo={fieldprops as any} />
      </Provider>,
    );
  });

  const EditComponent = (props: any) => (
    <Provider store={store}>
      <UserEditModal show onClose={() => false} mode="Edit" userInfo={fieldprops as any} />
    </Provider>
  );

  it('should match the snapshot', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('renders a <TextField/> and <Button /> component', () => {
    const Wrapper = mount(<EditComponent />);
    expect(Wrapper.find(TextField)).toHaveLength(3);
    expect(Wrapper.find(Button)).toHaveLength(2);
  });
});
