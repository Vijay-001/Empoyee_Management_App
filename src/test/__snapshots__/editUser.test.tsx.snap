// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminLogin/> should match the snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <UserEditModal
    mode="Edit"
    onClose={[Function]}
    show={true}
    userInfo={
      Object {
        "avatar": "",
        "email": "",
        "first_name": "",
        "id": "",
        "last_name": "",
        "password": "",
      }
    }
  />
</ContextProvider>
`;
